### ISSUE TITLE
Upgrade existing coin with Monero Generator

### ISSUE THREAD

**Issue by nidscom** [forknote/monero-generator#1]
Hi. Is it possible to do this? I have cryptonote coin that i would like to improve the code. RingCT would be awesome.

**Comment by pmitchev** [comment]
Yes. The work on this will be resumed soon.

PS are you having a bytecoin clone?

**Comment by nidscom** [comment]
Yes i forked from Cryptonotefoundation repo. But having problem with json rpc.

**Comment by RocksteadyTC** [comment]
We are a bytecoin clone, TurtleCoin. What do you have in mind @pmitchev

**Comment by pmitchev** [comment]
@RocksteadyTC The big goal of the project is to make all Forknote coins work on both Bytecoin and Monero cores.  The start for the project is to get Dashcoin working on Monero core, since it's the simplest case.  I got stuck at syncing V2 blocks, since bytecoin has some change in the block template. If you are willing to help this will be huge for small Cryptonote coins.

**Comment by RocksteadyTC** [comment]
@pmitchev I'd like to float this idea by our development team if you don't mind. This is a very promising opportunity in my opinion. They may feel differently, but I think they'd agree.

**Comment by Dreadmon** [comment]
@RocksteadyTC @pmitchev   I would like to get involved with the C++ coding.

**Comment by pmitchev** [comment]
@Dreadmon Hi,
The current status of this project: not able to sync after blocks with major version 2. As mentioned, the goals is to make it running for Dashcoin. From there, it's a simple config change to make it run to all others. 

The work done so far is here:
https://github.com/forknote/monero-generator/blob/master/extensions/bytecoin-clone.json

This section here is certainly flawed:
https://github.com/forknote/monero-generator/blob/master/extensions/bytecoin-clone.json#L233-L460

To generate the output source code:
bash generator.sh -j configs/dashcoin.json

Milestones: 
1) Able to sync (Dashcoin)
2) Able to mine (Dashcoin)
3) Adding tests

**Comment by Dreadmon** [comment]
@Dreadmon @RocksteadyTC   Thanks!   I will get started on this.

**Comment by Dreadmon** [comment]
@pmitchev  When I run the generator for dashcoin, I am getting this compilation below. 

Do you want me to fix this?  Or, I am missing something? 
Specifically:   **undefined reference to `rl_filename_completion_function'**


/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_f4f76.dir/link.txt --verbose=1
/usr/bin/cc  -fno-strict-aliasing -maes -std=c11 -D_GNU_SOURCE   -Wall -Wextra -Wpointer-arith -Wundef -Wvla -Wwrite-strings -Wno-error=extra -Wno-error=deprecated-declarations -Wno-unused-parameter -Wno-unused-variable -Wno-error=unused-variable -Wno-error=undef -Wno-error=uninitialized -Wlogical-op -Wno-error=maybe-uninitialized -Wno-error=cpp -Waggregate-return -Wnested-externs -Wold-style-definition -Wstrict-prototypes -march=x86-64  -fPIC  -Wformat -Wformat-security -fstack-protector -fstack-protector-strong -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=rl_filename_completion_function    -rdynamic CMakeFiles/cmTC_f4f76.dir/CheckFunctionExists.c.o  -o cmTC_f4f76
CMakeFiles/cmTC_f4f76.dir/CheckFunctionExists.c.o: In function `main':
CheckFunctionExists.c:(.text+0x10): undefined reference to `rl_filename_completion_function'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_f4f76.dir/build.make:97: recipe for target 'cmTC_f4f76' failed
make[2]: *** [cmTC_f4f76] Error 1
make[2]: Leaving directory '/home/scotty/forknote/monero-generator/generated_files/dashcoin.monero/build/release/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_f4f76/fast' failed
make[1]: *** [cmTC_f4f76/fast] Error 2
make[1]: Leaving directory '/home/scotty/forknote/monero-generator/generated_files/dashcoin.monero/build/release/CMakeFiles/CMakeTmp'

**Comment by RocksteadyTC** [comment]
@Dreadmon This might need a new issue
